File: backend/Core/Fast.Iaas/Internal/AppEnvironmentEnum.cs
namespace Fast.ServiceCollection.Internal;

/// <summary>
/// App运行环境枚举
/// </summary>
[FastEnum("App运行环境枚举")]
[Flags]
public enum AppEnvironmentEnum
{
    /// <summary>
    /// PC
    /// </summary>
    [Description("PC")]
    PC = 1,

    /// <summary>
    /// Windows端
    /// </summary>
    [Description("Windows")]
    Windows = 2,

    /// <summary>
    /// App端
    /// </summary>
    [Description("App")]
    App = 4,

    /// <summary>
    /// H5
    /// </summary>
    [Description("H5")]
    H5 = 8,

    /// <summary>
    /// 微信小程序
    /// </summary>
    [Description("微信小程序")]
    WeChatMiniProgram = 16,
}

File: frontend-template/src/api/enums/app-environment-enum.ts
/**
 *
 * @export
 * @enum {string}
 */
export enum AppEnvironmentEnum {
    /**
     * 网页
     */
    Web = 1,
    /**
     * Pc
     */
    PC = 2,
    /**
     * 微信小程序
     */
    WeChatMiniProgram = 4,
    /**
     * 安卓App
     */
    AndroidApp = 8,
    /**
     * IOSApp
     */
    IOSApp = 16,
    /**
     * 其他
     */
    Other = 512,
}
